CMakeLists.txt

```diff
@@ -69,6 +69,11 @@ add_subdirectory(dstulib)
 add_subdirectory(engine)
 add_subdirectory(keylib)
 
+target_include_directories(dstulib PRIVATE ${OPENSSL_INCLUDE_DIR})
+
+find_library(ICONV_LIBRARY NAMES iconv libiconv)
```

Comment:
This breaks build on Linux. There is no separate `libiconv` on Linux, it is included in `glibc`.

Consider using `find_package(Iconv)` and imported targets.

Reply:
Replaced with `find_package(Iconv)`. Moved this from root `CMakeList.txt` into `keylib/CMakeList.txt`.

Should I wrap it to be included only for MacOS?

```
if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
    find_package(Iconv REQUIRED)
    target_link_libraries(keylib PUBLIC ${ICONV_LIBRARY})
endif()
```

Reply:
If you use `find_package(Iconv)` with imported target `target_link_libraries(keylib PUBLIC Iconv::Iconv)`, it should work fine for both Linux and OSX.

Reply:
Replaced `${ICONV_LIBRARY}` with `Iconv::Iconv`